src/App.tsx

```diff
@@ -15,48 +16,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+type Goods = Array<string>;
+
+interface FnArg {
+  sortField: string;
+  reversed: boolean;
+}
+
+enum SortType {
+  sortByAlpha = 'alpha',
+  sortByLength = 'length',
+}
+
+function getPrepearedGoods(goods: Goods, { sortField, reversed }: FnArg) {
+  let prepearedGoods = [...goods];
+
+  if (sortField) {
+    prepearedGoods.sort((a, b) => {
+      switch (sortField) {
+        case SortType.sortByAlpha:
+          return a.localeCompare(b);
+        case SortType.sortByLength:
+          return a.length - b.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    prepearedGoods = prepearedGoods.reverse();
+  }
+
+  return prepearedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [reversed, setReversed] = useState(false);
+  const visibleGoods = getPrepearedGoods(goodsFromServer,
+    { sortField, reversed });
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
+          onClick={() => setSortField(SortType.sortByAlpha)}
           type="button"
-          className="button is-info is-light"
+          className={cn('button is-info',
+            { 'is-light': sortField !== SortType.sortByAlpha })}
         >
           Sort alphabetically
         </button>
 
         <button
+          onClick={() => setSortField(SortType.sortByLength)}
           type="button"
-          className="button is-success is-light"
+          className={cn('button is-success',
+            { 'is-light': sortField !== SortType.sortByLength })}
         >
           Sort by length
         </button>
 
         <button
+          onClick={() => setReversed(!reversed)}
           type="button"
-          className="button is-warning is-light"
+          className={cn('button is-warning', { 'is-light': !reversed })}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortField || reversed) && (
+          <button
+            onClick={() => {
+              setSortField('');
+              setReversed(false);
+            }}
```

Comment:
Move to separate function  (transfer the function to a variable, and here only write the name of the function)